src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -439,8 +468,8 @@ public Project updateProject(UUID uuid, String name, String description, String
         project.setVersion(version);
         project.setPurl(purl);
 
-        if (project.isActive() && hasActiveChild(project) && !active){
-            LOGGER.warn(project + " cannot be set to inactive, if active children are present. " + project + " remains active");
+        if (!active && project.isActive() && hasActiveChild(project)){
+            throw new IllegalArgumentException("Project cannot be set to inactive, if active children are present.");
         } else {
             project.setActive(active);
         }
```

Comment:
```suggestion
        }
        project.setActive(active);
```
Kannst dir das else sparen wenn im if eine exception fliegt